src/Products/PluginIndexes/common/UnIndex.py

```diff
@@ -324,6 +329,65 @@ def _apply_not(self, not_parm, resultset=None):
     def _convert(self, value, default=None):
         return value
 
+    def getRequestCache(self):
+        """returns dict for caching per request for interim results
+        of an index search. Returns 'None' if no REQUEST attribute
+        is available"""
+        cache = None
+        REQUEST = aq_get(self, 'REQUEST', None)
+        if REQUEST is not None:
+            catalog = aq_parent(aq_parent(aq_inner(self)))
+            if catalog is not None:
+                key = self._catalog_cache_key(catalog)
+                cache = REQUEST.get(key, None)
+                if cache is None:
+                    cache = REQUEST[key] = RequestCache()
+
+        return cache
+
+    def _catalog_cache_key(self, catalog):
```

Comment:
Since this is only called from inside getRequestCache, I'd inline it into that function. Since the method on its own is only dependent on the catalog object but not the index, it otherwise feels like it's in the wrong place and should have been a method of the catalog. By simply inline it, that concern goes away.